energy-integration/common-modules/weekly-rewards-splitting/src/update_claim_progress_energy.rs

```diff
@@ -73,6 +73,27 @@ pub trait UpdateClaimProgressEnergyModule:
         self.emit_update_user_energy_event(user, current_week, current_energy);
     }
 
+    fn clear_user_energy(&self, user: &ManagedAddress) {
```

Comment:
make this receive the min_farm_amount - and resolve the if here. You have duplicated code otherwise.

Reply:
Done.